src/include/storage/sql_table.h

```diff
@@ -140,93 +137,229 @@ class SqlTable {
    * output buffer. The tuples materialized are guaranteed to be visible and valid, and the function makes best effort
    * to fill the buffer, unless there are no more tuples. The given iterator is mutated to point to one slot past the
    * last slot scanned in the invocation.
-   *
+   * @warning a tuple inserted early than start_pos might appear in the scan (by migrating into a later version schema
+   * datatable so the ordering of the scanning is not strictly defined here. However, if start_pos is obtained through
+   * SqlTable::begin(), the results will always contain the same tuples
    * @param txn the calling transaction
-   * @param start_pos iterator to the starting location for the sequential scan
+   * @param start_pos iterator to the starting location for the sequential scan.
    * @param out_buffer output buffer. The object should already contain projection list information. This buffer is
    *                   always cleared of old values.
+   * @param layout_version Schema version the current querying transaction should see
+   */
+  void Scan(common::ManagedPointer<transaction::TransactionContext> txn, DataTable::SlotIterator *start_pos,
+            ProjectedColumns *out_buffer, layout_version_t layout_version = layout_version_t{0}) const;
```

Comment:
Shouldn't layout_version be a const?

Reply:
const is pointless when are passing arguments as values